framework/src/modules/dpos_v2/commands/change_commission.ts

```diff
@@ -0,0 +1,109 @@
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { validator } from '@liskhq/lisk-validator';
+import {
+	CommandVerifyContext,
+	VerificationResult,
+	VerifyStatus,
+	CommandExecuteContext,
+} from '../../../state_machine';
+import { BaseCommand } from '../../base_command';
+import { CommissionChangeEvent } from '../events/commission_change';
+import { changeCommissionCommandParamsSchema } from '../schemas';
+import { DelegateStore } from '../stores/delegate';
+import { ChangeCommissionParams } from '../types';
+
+export class ChangeCommissionCommand extends BaseCommand {
+	public schema = changeCommissionCommandParamsSchema;
+
+	private _commissionIncreasePeriod!: number;
+	private _maxCommissionIncreaseRate!: number;
+
+	public init(args: { commissionIncreasePeriod: number; maxCommissionIncreaseRate: number }) {
+		this._commissionIncreasePeriod = args.commissionIncreasePeriod;
+		this._maxCommissionIncreaseRate = args.maxCommissionIncreaseRate;
+	}
+
+	public async verify(
+		context: CommandVerifyContext<ChangeCommissionParams>,
+	): Promise<VerificationResult> {
+		try {
+			validator.validate(changeCommissionCommandParamsSchema, context.params);
+		} catch (err) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: err as Error,
+			};
+		}
+
+		const delegateStore = this.stores.get(DelegateStore);
+		const delegateExists = await delegateStore.has(context, context.transaction.senderAddress);
+
+		if (!delegateExists) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Transaction sender has not registered as a delegate.'),
+			};
+		}
+
+		const delegate = await delegateStore.get(context, context.transaction.senderAddress);
+		const oldCommission = delegate.commission;
+		const hasIncreasedCommissionRecently =
+			context.header.height - delegate.lastCommissionIncreaseHeight <
+			this._commissionIncreasePeriod;
+
+		if (context.params.newCommission > oldCommission && hasIncreasedCommissionRecently) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error(
+					`Can only increase the commission again ${this._commissionIncreasePeriod} blocks after the last commission increase.`,
+				),
+			};
+		}
+
+		if (context.params.newCommission - oldCommission > this._maxCommissionIncreaseRate) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error(
+					`Invalid argument: Commission increase larger than ${this._maxCommissionIncreaseRate}.`,
+				),
+			};
+		}
+
+		return {
+			status: VerifyStatus.OK,
+		};
+	}
+
+	public async execute(context: CommandExecuteContext<ChangeCommissionParams>): Promise<void> {
+		const delegateStore = this.stores.get(DelegateStore);
+
+		const delegate = await delegateStore.get(context, context.transaction.senderAddress);
+		const oldCommission = delegate.commission;
+
+		delegate.commission = context.params.newCommission;
+		if (delegate.commission > oldCommission) {
```

Comment:
```suggestion
		if (delegate.commission >= oldCommission) {
```

Reply:
also test related to this should check this

Reply:
I see now that the LIP also uses `>=` instead of `>`. But what's the point? If the requested new commission is the SAME as the existing commission, is it really a commission CHANGE? If you put on the same socks you were wearing yesterday, have you really changed your socks? 😀

And if we agree that there is no change in commission [and presume that the command was issued by mistake], what's the reason to update the value in `lastCommissionIncreaseHeight`? 🤔 

Imagine a scenario where a delegate wants to increase the commission, but by mistake they set the same commission. The system will update the delegate store. And now they will have to wait until the increase lockout period is over, to set the commission they really wanted.

If I was a validator, that would have made me very grumpy.

Reply:
This is a protocol change, so if you disagree we should discuss with research team, but I agree that in the equal scenario, we shouldn't update the lastCommissionIncreaseHeight. Could you raise it with @gkoumout ?

Reply:
> if a delegate increases the commission by 0, it's like a promise that they will not increase it for the next month. This way delegates kind of provide security to their voters.

By @gkoumout